fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/qrcode/QRCodeScanningViewModel.java

```diff
@@ -5,4 +5,6 @@ public interface QRCodeScanningViewModel extends QRCodeListener {
   int getScanDescription();
 
   ScanningAction getScanningAction();
+
+  boolean addManually(String data);
```

Comment:
why do we use that ?

Reply:
Fundamentally, it could be sold as an accessibility thing i.e. not everyone has a working camera. So it does not have absolutely no purpose user wise. The real reason it is added now though is for integration tests, where mocking the camera has proven almost impossible